lib/api/tensor-impl.ts

```diff
@@ -67,18 +66,17 @@ export class Tensor implements TensorInterface {
       if (dim >= this.dims[idx]) {
         throw new RangeError(`Input index array dims don't match the tensor dims.`);
       }
-      flatIndices += idx < indexArray.length - 1 ? dim * this.dims.slice(idx + 1).reduce((a, b) => a * b) : dim;
```

Comment:
So previously - we were doing the reduce every time instead of pre-computing strides at init time ? 

Currently, I am guessing when we `get()` from internalTensor, we will fetch the value with lesser overhead (i.e.) using strides ?

Reply:
yes, you are right. current internal tensor uses cache for strides (it will be only calculated once)

but this is still slow. in general, get() and set() are much less efficient than using the typed array directly.